server/cmwell-data-tools/src/test/scala/cmwell/tools/data/downloader/consumer/ConsumerSpec.scala

```diff
@@ -40,6 +40,30 @@ class ConsumerSpec extends BaseWiremockSpec {
     super.afterAll()
   }
 
+  it should "be reslient against server errors" in {
+
+    val exitRetry = "exit-retry"
+
+    stubFor(get(urlPathMatching("/.*")).inScenario(scenario)
+      .whenScenarioStateIs(Scenario.STARTED)
+      .willReturn(aResponse()
+        .withStatus(StatusCodes.OK.intValue)
+        .withHeader(CMWELL_POSITION, "3AAAMHwv"))
+      .willSetStateTo(exitRetry)
+    )
+
+    Downloader.createTsvSource(baseUrl = s"localhost:${wireMockServer.port}")
+
+    1 shouldBe 1
```

Comment:
You should count the number of times the wire mock server received an incoming request to `create-consumer`.
An assertion should verify that number of defined retries == actual number of incoming requests.

Reply:
New unit test added and working.